service/resource/legacy/daemonset.go

```diff
@@ -3,38 +3,44 @@ package legacy
 import (
 	"github.com/giantswarm/flanneltpr"
 	apismeta "k8s.io/apimachinery/pkg/apis/meta/v1"
-	apisextensions "k8s.io/client-go/pkg/apis/extensions/v1beta1"
-
 	"k8s.io/apimachinery/pkg/util/intstr"
 	api "k8s.io/client-go/pkg/api/v1"
+	apisextensions "k8s.io/client-go/pkg/apis/extensions/v1beta1"
+
+	"github.com/giantswarm/flannel-operator/service/key"
 )
 
-func newDaemonSet(spec flanneltpr.Spec, etcdCAFile, etcdCrtFile, etcdKeyFile string) *apisextensions.DaemonSet {
+func newDaemonSet(customObject flanneltpr.CustomObject, etcdCAFile, etcdCrtFile, etcdKeyFile string) *apisextensions.DaemonSet {
```

Comment:
Since this is the legacy resource there is still a lot left to do. I tried to minimize the changes as much as possible while trying to pave the ground for "a better future".